trieste/data.py

```diff
@@ -202,3 +202,46 @@ def add_fidelity_column(query_points: TensorType, fidelity: int) -> TensorType:
     fidelity_col = tf.ones((tf.shape(query_points)[-2], 1), dtype=query_points.dtype) * fidelity
     query_points_for_fidelity = tf.concat([query_points, fidelity_col], axis=-1)
     return query_points_for_fidelity
+
+
+def split_dataset_randomly(
+    dataset: Dataset, proportion: float, seed: Optional[int] = None
+) -> tuple[Dataset, Dataset]:
+    """Split dataset randomly into two parts according to the specified proportion.
+
+    :param dataset: Dataset to split.
+    :param proportion: Proportion of data to include in first dataset (between 0 and 1).
+    :param seed: Optional random seed for reproducibility.
+    :return: Tuple of two datasets, where first contains approximately proportion * len(dataset)
+        points.
+    :raise ValueError: If proportion is not between 0 and 1
+    """
+    if not 0 <= proportion <= 1:
+        raise ValueError(f"proportion must be between 0 and 1, got {proportion}")
+
+    if len(dataset) == 0:
+        return dataset, dataset
+
+    if seed is not None:  # ensure reproducibility
```

Comment:
I think this can be removed now that we're passing the seed into `tf.random.shuffle`?